### ISSUE TITLE
Make Visitor public class

### ISSUE THREAD

**Issue by mst-avaleo** [everit-org/json-schema#376]
I've started to use the json-schema library for other purpose than validation. My goal is to analyse JSON schemas from different services in our organization and enforce some internal rules. In order to do it I was looking for a library, which will help parse JSON schemas (load references, ensure correct defaults etc). I haven't found any generic json schema parser, therefore I've started to look on json validation tools. After trying out few, I've found this library which gives me what I need. 

The only thing I don't like is the way how I'm forced to traverse the json schema tree - it forces to do plenty of `instanceof`s. Though I've seen that the validation and toString is implemented as Visitor pattern, which would be great also for my use case. Unfortunately I cannot use it, since the `Visitor` class and `accept()` methods are package protected.

What do you think about making `Visitor` pattern available as API?

**Comment by erosb** [comment]
> I'm forced to traverse the json schema tree

yes, I suggest taking that approach. The visitor API is intentionally package-private. It has some flaws, I'm not particularly happy with it. As long as it isn't public, there is chance that it will be fixed up without needing to maintain backward-compatibility.

Also please be aware that this library is very low on development resources this year, there is no ETA for draft-2019-09 (contributions are welcome though, there is some initial work done in the ` draft-2019-09` branch).
